core/templates/dev/head/pages/exploration_player/SupplementalCardDirective.js

```diff
@@ -90,6 +120,7 @@ oppia.directive('supplementalCard', [
 
           $scope.$on(EVENT_ACTIVE_CARD_CHANGED, function() {
             updateActiveCard();
+            HintManagerService.reset();
```

Comment:
Still not sure I understand this. When is EVENT_ACTIVE_CARD_CHANGED fired? Why isn't this taken care of during init()?